python/tvm/topi/unique.py

```diff
@@ -0,0 +1,118 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint: disable=invalid-name
+"""Unique operator"""
+from ..te import hybrid
+from .cumsum import cumsum
+from .sort import sort, argsort
+
+
+@hybrid.script
+def _calc_adjacent_diff(data):
+    output = output_tensor(data.shape, "int32")
+    output[0] = int32(0)
+    for i in range(1, data.shape[0]):
+        output[i] = int32(1) if data[i] != data[i - 1] else int32(0)
+    return output
+
+
+@hybrid.script
+def _calc_num_unique(data):
+    output = output_tensor((1,), "int32")
+    output[0] = data[data.shape[0] - 1] + 1
+    return output
+
+
+@hybrid.script
+def _calc_unique_sorted(data, argsorted_indices, inc_scan):
+    unique_elements = output_tensor(data.shape, data.dtype)
+    indices = output_tensor(data.shape, "int32")
+    for i in range(data.shape[0]):
+        indices[argsorted_indices[i]] = inc_scan[i]
```

Comment:
We can do this loop in parallel

Reply:
actually all loops can be done in parallel, I changed all loops to parallel.